mypyc/irbuild/specialize.py

```diff
@@ -356,3 +357,45 @@ def translate_dict_setdefault(
                               [callee_dict, key_val, data_type],
                               expr.line)
     return None
+
+
+@specialize_function('format', str_rprimitive)
+def translate_str_format(
+        builder: IRBuilder, expr: CallExpr, callee: RefExpr) -> Optional[Value]:
+    if isinstance(callee, MemberExpr) and isinstance(callee.expr, StrExpr):
+        # TODO
+        # Only consider simplest situation here
+        format_str = callee.expr.value
+        if not str_format_check_helper(format_str):
+            return None
+
+        literals = [builder.load_str(x) for x in format_str.split("{}")]
+        variables = [builder.call_c(str_op, [builder.accept(x)], expr.line) for x in expr.args]
+
+        # Allocate space for the parameter list of CPyStr_Build().
+        result_len = len(literals) + len(variables)
+        result_list: List[Value] = [Integer(0, c_int_rprimitive)] * (result_len + 1)
+
+        # The first parameter is the total size of the following PyObject* merged from
+        # two lists alternatively.
+        result_list[0] = Integer(result_len, c_int_rprimitive, expr.line)
+        result_list[1::2] = literals
+        result_list[2::2] = variables
```

Comment:
These slice assignments are somewhat difficult to reason about (a bit too clever, I think). I'd prefer a simple for loop, possible with `zip()`. This way you should be able to build the list incrementally using `append` instead of first creating it and then mutating it. Appending to a list is more Pythonic than creating a list first with dummy values and then assigning to it.